src/Http/Http/src/Internal/DefaultHttpResponse.cs

```diff
@@ -70,8 +66,26 @@ public override IHeaderDictionary Headers
 
         public override Stream Body
         {
-            get { return HttpResponseFeature.Body; }
-            set { HttpResponseFeature.Body = value; }
+            get { return HttpResponseBodyFeature.Stream; }
+            set
+            {
+                var otherFeature = _features.Collection.Get<IHttpResponseBodyFeature>();
+                if (otherFeature is StreamResponseBodyFeature streamFeature
+                    && streamFeature.PriorFeature != null
+                    && object.ReferenceEquals(value, streamFeature.PriorFeature.Stream))
+                {
+                    // They're reverting the stream back to the prior one. Revert the whole feature.
+                    _features.Collection.Set(streamFeature.PriorFeature);
+                    // CompleteAsync is registered with HttpResponse.OnCompleted and there's no way to unregister it.
+                    // Prevent it from running by marking as disposed.
+                    streamFeature.Dispose();
+                    return;
+                }
+
+                var feature = new StreamResponseBodyFeature(value, otherFeature);
+                OnCompleted(feature.CompleteAsync);
```

Comment:
I'm having second thoughts about registering `OnCompleted` here. When you revert to the older feature, there's no way to undo this. This is an issue with the fact that you can't un-register the callback when using the On* callbacks.

Reply:
Ah I just read the comment 🤣